sway-lib-std/src/fixed_point.sw

```diff
@@ -0,0 +1,130 @@
+library fixed_point;
+//! A wrapper around U128 type for a library for Sway for mathematical functions operating with signed 64.64-bit fixed point numbers. 
+
+use ::u128::U128;
+use ::assert::assert;
+
+pub struct UFP64 {
+    value: U128 
+}
+
+impl core::ops::Eq for UFP64 {
+    pub fn eq(self, other: Self) -> bool {
+        self.value == other.value
+    }
+}
+
+impl core::ops::Ord for UFP64 {
+    pub fn gt(self, other: Self) -> bool {
+        self.value > other.value
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.value < other.value
+    }
+}
+
+impl core::ops::Add for UFP64 {
+    /// Add a UFP64 to a UFP64. Panics on overflow.
+    pub fn add(self, other: Self) -> Self {
+        UFP64 {
+            value: self.value + other.value
+        }
+    }
+}
+
+impl core::ops::Subtract for UFP64 {
+    /// Subtract a UFP64 from a UFP64. Panics of overflow.
```

Comment:
Same here, isn't this an "underflow" instead of an "overflow"?

Reply:
This is actually also an overflow. Underflow is only when the floating-point result has too many decimals, not when it's too negative.